source/io/module_javaScriptInvoke.js

```diff
@@ -153,198 +264,85 @@
-                break;
-            case 'Int16':
-                returnTypeMistmatch = updateInt16ReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'Int32':
-                returnTypeMistmatch = updateInt32ReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'UInt8':
-                returnTypeMistmatch = updateUInt8ReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'UInt16':
-                returnTypeMistmatch = updateUInt16ReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'UInt32':
-                returnTypeMistmatch = updateUInt32ReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'Single':
-                returnTypeMistmatch = updateSingleReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'Double':
-                returnTypeMistmatch = updateDoubleReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'String':
-                returnTypeMistmatch = updateStringReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'Boolean':
-                returnTypeMistmatch = updateBooleanReturnValue(javaScriptReturnTypeName, returnValuePointer, returnValue);
-                break;
-            case 'StaticTypeAndData': {
-                break;
-            }
-            default: {
-                newErrorStatus = true;
-                newErrorCode = ERRORS.kNIUnsupportedLabVIEWReturnTypeInJavaScriptInvoke.CODE;
-                newErrorSource = ERRORS.kNIUnsupportedLabVIEWReturnTypeInJavaScriptInvoke.MESSAGE + '\'' + functionNameString + '\'.';
-                Module.coreHelpers.mergeErrors(newErrorStatus, newErrorCode, newErrorSource, errorStatusPointer, errorCodePointer, errorSourcePointer);
-                return;
-            }
-            }
 
-            if (returnTypeMistmatch) {
-                newErrorStatus = true;
-                newErrorCode = ERRORS.kNITypeMistmatchForReturnTypeInJavaScriptInvoke.CODE;
-                newErrorSource = ERRORS.kNITypeMistmatchForReturnTypeInJavaScriptInvoke.MESSAGE + '\'' + functionNameString +
-                    '\'. JavaScript return type is \'' + javaScriptReturnTypeName +
-                    '\'. LabVIEW return type is \'' + returnTypeName + '\'';
-                Module.coreHelpers.mergeErrors(newErrorStatus, newErrorCode, newErrorSource, errorStatusPointer, errorCodePointer, errorSourcePointer);
+            try {
+                updateReturnValueGivenType(returnTypeName, returnValuePointer, returnValue);
```

Comment:
.... here. This is the second time we check if it is a a valid JS return type. From what I see in the function `updateReturnValueGivenType()` the code is added as the name of the exception but I don't see it being used here.

Reply:
I think we can get rid of the `isValidJavaScriptReturnType` function as well as `isTypedArray` since we now have the `isValidReturnType` family of functions declared in `functionTypes`. Basically would be the work of moving the validation in `updateReturnValueGivenType()` outside of it and validating before calling it.